lib/import/brca/providers/st_george/st_george_handler.rb

```diff
@@ -1,436 +1,360 @@
+              genotype.add_test_scope(:no_genetictestscope)
             end
+            genotype
           end
-          # rubocop:enable  Lint/DuplicateBranch
 
-          def process_single_protein(genotype, record)
-            if record.raw_fields['genotype'].scan(PROTEIN_REGEX).size.positive?
-              genotype.add_protein_impact($LAST_MATCH_INFO[:impact])
-              @logger.debug "SUCCESSFUL gene parse for: #{$LAST_MATCH_INFO[:impact]}"
-            else
-              @logger.debug "FAILED protein parse for: #{record.raw_fields['genotype']}"
-            end
-          end
+          def process_genes_targeted(record)
+            # For targeted tests only
+            # This method creates a list of genes included in the record that match BRCA_GENE_REGEX
+            # The genotype is duplicated for each gene in this list
+            # A list of genotypes (one for each gene) is returned
 
-          def process_single_positive_variants(genotype, record)
-            if positive_cdna?(record)
-              process_cdna_variant(genotype, record)
-            elsif positive_exonvariant?(record)
-              process_exonic_variant(genotype, record)
-            else
-              @logger.debug "FAILED variant parse for: #{record.raw_fields['genotype']}"
-            end
-          end
+            columns = ['gene', 'gene (other)']
+            genes = []
+            columns.each do |column|
+              gene_list = record.raw_fields[column]&.scan(BRCA_GENE_REGEX)
+              next if gene_list.blank?
 
-          def add_variants_multiple_results(variants, genotype, genotypes)
-            variants.each do |gene, mutation, protein|
-              genotype_dup = genotype.dup
-              genotype_dup.add_gene(gene)
-              genotype_dup.add_gene_location(mutation)
-              genotype_dup.add_protein_impact(protein)
-              genotype_dup.add_status(2)
-              genotypes.append(genotype_dup)
+              gene_list.each do |gene|
+                gene = BRCA_GENE_MAP[gene]
+                genes.append(gene)
+              end
             end
+            genes
           end
 
-          def process_multiple_positive_variants(positive_genes, genotype, record, genotypes)
-            if positive_genes.flatten.uniq.size > 1
-              variants = process_multi_genes_rec(record, positive_genes)
-            elsif positive_genes.flatten.uniq.size == 1
-              variants = process_uniq_gene_rec(record, positive_genes)
+          def duplicate_genotype_targeted(genes, genotype)
+            # When there is more than one gene listed a separate genotype needs to be created for each one
+            # The genotype is duplicated and the new gene is added to the duplicated genotype
+            # Each genotype is then added to the genoytypes list which this method then returns
+            genotypes = []
+            genes.each do |gene|
```

Comment:
Not very sure with two loops logic here.. can methods like genes.flatten.compact_blank.uniq be used to reduce it to single genes array to loop?

Reply:
see [b72975c](https://github.com/NHSDigital/data_management_system/pull/77/commits/b72975c9bb7ce4cd2ac1e0fa3bd23ff936a97093)